### ISSUE TITLE
Generator turns `type` properties to `_type`, but `type` is a valid Swift name

### ISSUE THREAD

**Issue by MahdiBM** [apple/swift-openapi-generator#402]
The same as the title:
The generator turns properties named `type` to `_type`, but `type` is a valid Swift name.

**Comment by simonjbeaumont** [comment]
I have a feeling we did this for a reason. Potentially it was ambiguity with `type(of:)`. This might have been before we started using fully qualified names in as many places as possible.

In any event, I'd rather be overly cautious here and leave the behaviour as it is.

If it turns out we _were_ overly cautious, the only harm is that operations, schemas, and parameters with this one name are unnecessarily "safed". 

If it turns out there are some pathological OpenAPI documents that cause a breakage then we'll need to make a fix which will then be API breaking.
